src/coord/src/catalog.rs

```diff
@@ -2119,7 +2153,7 @@ impl Catalog {
         Ok(temporary_ids)
     }
 
-    pub fn transact<F, T>(
+    pub async fn transact<F, T>(
```

Comment:
funny that `F` didn't need to be async

do we expect people to do purely, basically cheap, data transformations in that callback?

Reply:
Yes. Currently this is used to do things like frob up a DataflowDesc that will be used later on to spin up a dataflow. And mostly that's it.